pkg/byteconv/byteconv.go

```diff
@@ -21,9 +22,17 @@ func ParseBool(b []byte) (bool, error) {
 }
 
 func ParseIP(b []byte) (net.IP, error) {
-	ip := net.ParseIP(UnsafeString(b))
+	unsafeStr := UnsafeString(b)
+	ip := net.ParseIP(unsafeStr)
 	if ip == nil {
-		return nil, fmt.Errorf("bad addr value field %v", b)
+		var err error
+		if utf8.Valid(b) {
+			// Print a better error message if b is a utf8 string
+			err = fmt.Errorf("Cannot parse %s as an IP address", unsafeStr)
```

Comment:
```suggestion
			err = fmt.Errorf("Cannot parse %s as an IP address", string(b))
```

Reply:
How about just
```go
return nil, fmt.Errorf("bad addr value field %q", b)
```
so we get the same representation for b used in errors from strconv?

Reply:
The issue raised by Phil (#927) was that "bad addr value field" followed by a byte array was confusing for end users who had supplied an IP address as part of a query. Another approach to fixing that would be to leave this function alone, as you suggest, and create another code path to validate IP addresses provided as query strings. I will defer to your preference as to whether to change this code, or add a separate validation function for user provided IP addr.

Reply:
My original comment wasn't very clear. I'm sorry about that.

I'm not suggesting we leave this function alone. It's definitely the right place to fix #927.

What I meant to suggest is that we change the %v verb in the current format specifier to %q, like this.
```diff
- return nil, fmt.Errorf("bad addr value field %v", b)
+ return nil, fmt.Errorf("bad addr value field %q", b)
```

%q ultimately invokes strconv.AppendQuote (https://github.com/golang/go/blob/go1.15/src/fmt/format.go#L444-L459), and that would give us the same representation used in errors from the strconv.Parse* functions (https://github.com/golang/go/blob/go1.15/src/fmt/format.go#L444-L459) that implement everything else in this file.

If we also want to change the "bad addr value field" string, I suggest something like "net.ParseIP: parsing %q: syntax error" for symmetry with strconv.NumError.Error.

Whatever we do decide, [avoid capitalizing the first word of the error string](https://github.com/golang/go/wiki/CodeReviewComments#error-strings).

Reply:
I'm late to this one because I don't manage to follow every dev PR. But I just wanted to say that while I respect the spirit of standardizing that @nwt has advocated here, let's not lose sight of the fact that users actually see these errors. Showing the errors with names of Go functions or following the Go standards of how error messages are formatted in Go context is surely helpful to us as Go developers, and also to community Go developers who start working on the project. On the other hand, security analysts that only know the Brim app may have zero coding/scripting experience and may benefit from error messages that are closer to natural language. Perhaps the ultimate solution would be to have alternate wordings for errors and a way to automatically deliver the appropriate one based on audience, e.g. whether they're an app user or a zq user, or whether they've got some preference set for "developer mode" etc. In any case, I'm not advocating we dig into any of that now, but figured I'd mention it while the topic is fresh in everyone's minds.